### DISCUSSION TITLE
If isMultiple=false, the data is still stored as for the multiple case

### DISCUSSION THREAD

**Discussion by Raistah** [awcodes/filament-curator#420] [Ideas]
# My case
I have this form with repeater, and each row shoud contain only one image
![photo_2024-01-12_19-14-10](https://github.com/awcodes/filament-curator/assets/29507215/988c78df-688c-4212-8840-04e9094c43c2)

But on frontend i still have array (of one item).
Also, since the keys of the array are uuid, I need to do a splice to get the first element

### default
![photo_2024-01-12_19-17-46](https://github.com/awcodes/filament-curator/assets/29507215/231f84b8-c6cc-4e02-8b4b-59af8c43e3f8)

### my way
![photo_2024-01-12_19-53-12](https://github.com/awcodes/filament-curator/assets/29507215/9b8c8c0b-f09a-4dd3-9fd1-a1c0b94c053f)

If there is already a way to do this and I'm did this for nothing (I'm still using filament v2), then show me because I haven't found it.

**awcodes\filament-curator\src\Components\Forms\CuratorPicker.php**
![photo_2024-01-12_20-00-59](https://github.com/awcodes/filament-curator/assets/29507215/941fd083-1482-4c0c-809d-4fa0d85f3c82)

**awcodes\filament-curator\resources\views\components\forms\picker.blade.php**
![photo_2024-01-12_20-01-02](https://github.com/awcodes/filament-curator/assets/29507215/577fa3f4-dc70-4cdc-90a7-3beb7fad2713)

It's a quick and not optimal solution, so I don't send a pull request, if it's not ready yet, I'll make it better for v3 later

version: ^2.7

**Comment by awcodes** [comment]
It's forced into an array to keep the data normalized and to mitigate a breaking change between v1 and v2. However, current release is v3 and work on v4 has started. 

You can also just use the Arr helper from laravel to not have to use the splice.

Personally I recommend doing all Curator media as a relationship. It just makes it all cleaner and you won't have to worry about this.

**Reply by Raistah** [reply]
Arr as an option, I hadn't thought about it, thanks.

I'm using an approach similar to the Fabricator plugin, where a page is made up of sections, i.e. pages can be "built" from the admin panel. The sections themselves are stored in the database as JSON, so a relationship is not really suitable.
